src/query/pipeline/sources/src/processors/sources/input_formats/impls/input_format_xml.rs

```diff
@@ -0,0 +1,254 @@
+                    let err_msg = format_column_error(schema, c, col_data, &e.message());
+                    return Err(xml_error(&err_msg, path, row_index));
+                };
+                reader.ignore_white_spaces().expect("must success");
+                if reader.must_eof().is_err() {
+                    let err_msg = format_column_error(schema, c, col_data, "bad field end");
+                    return Err(xml_error(&err_msg, path, row_index));
+                }
+            }
+            field_start = field_end;
+        }
+        Ok(())
+    }
+}
+
+impl InputFormatTextBase for InputFormatXML {
+    fn format_type() -> StageFileFormatType {
+        StageFileFormatType::Xml
+    }
+
+    fn get_format_settings(settings: &Arc<Settings>) -> Result<FormatSettings> {
+        let timezone = get_time_zone(settings)?;
+        Ok(FormatSettings {
+            row_tag: settings.get_row_tag()?.into_bytes(),
+            rowset_tag: settings.get_rowset_tag()?.into_bytes(),
+            timezone,
+            ..Default::default()
+        })
+    }
+
+    fn default_field_delimiter() -> u8 {
+        b','
+    }
+
+    fn deserialize(builder: &mut BlockBuilder<Self>, batch: RowBatch) -> Result<()> {
+        tracing::debug!(
+            "xml deserializing row batch {}, id={}, start_row={:?}, offset={}",
+            batch.path,
+            batch.batch_id,
+            batch.start_row,
+            batch.offset,
+        );
+        let columns = &mut builder.mutable_columns;
+        let n_column = columns.len();
+
+        let mut start = 0usize;
+        let mut field_end_idx = 0usize;
+        let start_row = batch.start_row.expect("must be success");
+        for (i, end) in batch.row_ends.iter().enumerate() {
+            let buf = &batch.data[start..*end];
+            Self::read_row(
+                buf,
+                columns,
+                &builder.ctx.schema,
+                &batch.field_ends[field_end_idx..field_end_idx + n_column],
+                &builder.ctx.format_settings,
+                &batch.path,
+                start_row + i,
+            )?;
+            start = *end;
+            field_end_idx += n_column;
+        }
+        Ok(())
+    }
+
+    fn align(state: &mut AligningState<Self>, buf: &[u8]) -> Result<Vec<RowBatch>> {
+        let xml_state = state.xml_reader.as_ref().expect("must be success");
+
+        let start_row = state.rows;
+        state.offset += buf.len();
+
+        let mut buf = buf;
+        let reader = ParserConfig::new().create_reader(&mut buf);
+
+        let mut rows_to_skip = state.rows_to_skip;
+
+        let mut output = RowBatch {
+            data: vec![],
+            row_ends: vec![],
+            field_ends: vec![],
+            path: state.path.to_string(),
+            batch_id: state.batch_id,
+            offset: 0,
+            start_row: Some(state.rows),
+        };
+
+        let mut cols = Vec::<Vec<u8>>::with_capacity(state.num_fields);
```

Comment:
should use a map here?

Reply:
there's no need to add a key here, so i'm not using a map.

Reply:
I mean order column names in XML may be different with that in table definination?

Reply:
i think there is a problem, herer we should not force the user to be defined in the same order as the corresponding table. 
😊 i'll fix it later.